TargetComposition/SymBuilder/SymBuilderTests/Program.cs

```diff
@@ -0,0 +1,447 @@
+            case System.Runtime.InteropServices.Architecture.Arm:
+                return Path.Combine(path, useArchName ? "x86" : "woa");
+            case System.Runtime.InteropServices.Architecture.Arm64:
+                return Path.Combine(path, "arm64");
+            default:
+                return path;
+        }
+    }
+
+    /// <summary>
+    /// As JsProvider.dll is not available in a NuGET package, we cannot pull it as we do with other engine
+    /// bits.  The tests here are in JavaScript and require it.  This will attempt to find an appropriate
+    /// debugger install from which to pull JsProvider without being manually pointed at such.
+    ///
+    /// Such can be either a Win10 SDK install (available externally) or a Microsoft internal ring install of
+    /// the debugger (available only internally).  We will not be able to pull from the store installed
+    /// version of WinDbgNext. 
+    /// 
+    /// Note that this can be overriden via the command line pointing at an appropriate install.
+    /// </summary>
+    /// <returns></returns>
+    static string? FindAppropriateDebuggerInstallPath()
+    {
+        string? installPath = null;
+
+        //
+        // Check for an internal ring install.
+        //
+        string internalPath = System.Environment.ExpandEnvironmentVariables("%LOCALAPPDATA%\\dbg\\UI");
+        string versionFile = Path.Combine(internalPath, "currentVersion.txt");
+        if (File.Exists(versionFile))
+        {
+            using (StreamReader reader = File.OpenText(versionFile))
+            {
+                if (reader.Peek() >= 0)
+                {
+                    string? line = reader.ReadLine();
+                    if (line != null)
+                    {
+                        installPath = AppendPlatformArch(Path.Combine(internalPath, line.Trim()), false);
+                    }
+                }
+            }
+        }
+
+        //
+        // Check for a Windows 10 SDK installation of the debugger.
+        //
+        if (installPath == null)
+        {
+            var installRootsKey = Microsoft.Win32.Registry.LocalMachine.OpenSubKey("SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots");
+            if (installRootsKey != null)
+            {
+                Object val = installRootsKey.GetValue("WindowsDebuggersRoot10");
+                if (val != null)
+                {
+                    installPath = AppendPlatformArch((string)val, true);
+                }
+            }
+        }
+
+        return installPath;
+    }
+    static void Main(string[] args)
+    {
+        string? debuggerInstallPath = FindAppropriateDebuggerInstallPath();
+
+        if (args.Length > 0)
+        {
+            debuggerInstallPath = args[0];
+        }
+
+        if (debuggerInstallPath == null)
+        { 
+            Console.WriteLine("Usage: SymbolBuilderTests <debugger install path> [<script path>]");

```

Comment:
><debugger install path> [<script path>] [](http://example.com/codeflow?start=57&length=39)

At some point, I'll expand upon this...  I'm curious as to thoughts though....  This whole notion seems like it would be more generally useful than as a tiny test harness for this particular project....  It's *REALLY* nice to be able to quickly drive / put together tests in JS.

If there's enough interest / value, maybe this *EVENTUALLY* makes sense as its own separate project that we pull into this one to run its specific unit tests...

Thoughts...?

Reply:
Agreed. I think it would be good to expose the project as a test adapter so the test cases could integrate with the rest of the build system.

Reply:
Can you clarify what you mean here by "test adapter" ...  to what infra...?

Reply:
A test adapter in .NET is a component that can provide test cases, test execution, and test results to a test runner. The normal test running is usually "dotnet test". The advantage of mapping to a test adapter is that you get some infra on top of it like better UI for test results or reports on test reliability.